File: guilib-bukkit/src/main/java/net/developertobi/guilib/bukkit/utils/SoundUtils.java
package net.developertobi.guilib.bukkit.utils;

import net.developertobi.guilib.bukkit.GuiLibBukkit;
import org.bukkit.Sound;
import org.bukkit.entity.Player;

public final class SoundUtils {

    private static Sound getClickSound() {
        Sound s = GuiLibBukkit.getInstance().getSoundConfigFile().getOnClick();
        return s != null ? s : Sound.BLOCK_NOTE_BLOCK_HAT;
    }

    private static Sound getOpenSound() {
        Sound s = GuiLibBukkit.getInstance().getSoundConfigFile().getOnOpen();
        return s != null ? s : Sound.BLOCK_COPPER_DOOR_OPEN;
    }

    private static Sound getCloseSound() {
        Sound s = GuiLibBukkit.getInstance().getSoundConfigFile().getOnClose();
        return s != null ? s : Sound.BLOCK_COPPER_DOOR_CLOSE;
    }

    private static Sound getPageSwitchSound() {
        Sound s = GuiLibBukkit.getInstance().getSoundConfigFile().getOnPageSwitch();
        return s != null ? s : Sound.BLOCK_COPPER_BULB_TURN_ON;
    }

    public static void playClickSound(Player player) {
        playSound(player, getClickSound());
    }

    public static void playOpenSound(Player player) {
        playSound(player, getOpenSound());
    }

    public static void playCloseSound(Player player) {
        playSound(player, getCloseSound());
    }

    public static void playSwitchPageSound(Player player) {
        playSound(player, getPageSwitchSound());
    }

    private static void playSound(Player player, Sound sound) {
        if (sound == null) return;
        player.playSound(player.getLocation(), sound, 1f, 1f);
    }
}

File: guilib-bukkit/bin/main/net/developertobi/inventorylib/bukkit/utils/SoundUtils.kt
package net.developertobi.inventorylib.bukkit.utils

import net.developertobi.inventorylib.bukkit.GuiInventoryBukkit
import org.bukkit.Sound
import org.bukkit.entity.Player

object SoundUtils {

    val CLICK: Sound? = GuiInventoryBukkit.instance.soundConfigFile.onClick ?: Sound.BLOCK_NOTE_BLOCK_HAT
    val OPEN: Sound? = GuiInventoryBukkit.instance.soundConfigFile.onOpen ?: Sound.BLOCK_COPPER_DOOR_OPEN
    val CLOSE: Sound? = GuiInventoryBukkit.instance.soundConfigFile.onClose ?: Sound.BLOCK_COPPER_DOOR_CLOSE
    val PAGE_SWITCH: Sound? = GuiInventoryBukkit.instance.soundConfigFile.onPageSwitch ?: Sound.BLOCK_COPPER_BULB_TURN_ON

    fun playClickSound(player: Player) {
        playSound(player, CLICK)
    }

    fun playOpenSound(player: Player) {
        playSound(player, OPEN)
    }

    fun playCloseSound(player: Player) {
        playSound(player, CLOSE)
    }

    fun playSwitchPageSound(player: Player) {
        playSound(player, PAGE_SWITCH)
    }

    private fun playSound(player: Player, sound: Sound?) {
        if (sound == null) return
        player.playSound(player.location, sound, 1f, 1f)
    }

}